src/settings.js

```diff
@@ -1203,17 +1203,6 @@ var BINARYEN_SCRIPTS = "";
 // codebase it may help reduce code size a little bit.
 var BINARYEN_IGNORE_IMPLICIT_TRAPS = 0;
 
-// How we handle wasm operations that may trap, which includes integer
-// div/rem of 0 and float-to-int of values too large to fit in an int.
-//   js: do exactly what js does. this can be slower.
-//   clamp: avoid traps by clamping to a reasonable value. this can be
-//          faster than "js".
-//   allow: allow creating operations that can trap. this is the most
-//          compact, as we just emit a single wasm operation, with no
-//          guards to trapping values, and also often the fastest.
-// [fastcomp-only]
-var BINARYEN_TRAP_MODE = "allow";
```

Comment:
I think rather than just remove these settings we should add them to `LEGACY_SETTINGS` so that more meaningful errors are produced?

Reply:
Oh, I pushed something earlier to ignore them, but re-reading it, this is one on which we errored actually.

I pushed something now to emit a proper error. Basically using the existing legacy settings, if you set an impossible value as the POSSIBLE_VALUES, it will always error. Is that good enough do you think, or should we add a completely new mechanism?

Reply:
I wonder what @juj would prefer here?   Perhaps he would prefer that when a setting is completely removed we just silently ignore it when not in `STRICT` mode.  I think the idea is that one can upgrade your SDK without having to modify your settings.

Perhaps just leave this part out of this PR so we can discuss separately?

Reply:
This issue will affect all other PRs that remove settings, though. I think it might be simplest to have these PRs have no functional change, that is, continue to error if we did before, and not if we didn't (as this PR now does). Then we can separately from all these PRs we can discuss later whether STRICT mode should be different?

Reply:
sgtm